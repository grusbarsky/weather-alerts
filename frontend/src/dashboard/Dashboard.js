import ArticleList from "../articles/ArticleList";
import UserArticleList from "../articles/UserArticleList";
import LocationList from "../locations/LocationList";
import AlertList from "../alerts/AlertList";
import React, { useContext } from "react";
import { Link } from "react-router-dom";
import UserContext from "../auth/UserContext";



function Dashboard() {
    const { currentUser } = useContext(UserContext);
    return (
        <div className="homepage">
            <div className="container text-center">
                <h1>Weather Alerts</h1>
                {!currentUser
                    ? (
                        <p>
                            <Link className="btn btn-primary m-2"
                                to="/login">
                                Log in
                            </Link>
                            <Link className="btn btn-primary m-2"
                                to="/signup">
                                Sign up
                            </Link>
                        </p>
                    ) : <div className='dashboard container'>
                        <div className='row'>
                            <div className='right col'>
                                <div className='my-locations card shadow my-3'>
                                    <h4 className='mt-4'>My Locations</h4>
                                    <LocationList />
                                </div>
                                <div className='my-alerts card shadow my-3'>
                                    <h4 className='mt-4'>My Alerts</h4>
                                    <AlertList width='w-75'/>
                                </div>
                                <div className='my-articles card shadow my-3'>
                                    <h4 className='mt-4'>My Articles</h4>
                                    <UserArticleList width='w-75'/>
                                </div>
                            </div>
                            <div className='col'>
                                <div className='articles left card shadow my-3 mh-25'>
                                    <h4 className='mt-4'>News</h4>
                                    <ArticleList width='w-75' />
                                </div>
                            </div>
                        </div>
                    </div>
                }
            </div>
        </div>
    )
}

export default Dashboard;