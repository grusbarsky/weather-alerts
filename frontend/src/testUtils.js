import React from "react";
import UserContext from "./auth/UserContext";


const testUser = {
  username: "testuser",
  first_name: "testfirst",
  last_name: "testlast",
  email: "test@test.net",
  enableAlerts: false,
};

const UserProvider =
    ({ children, currentUser = testUser}) => (
    <UserContext.Provider value={{ currentUser }}>
      {children}
    </UserContext.Provider>
);

export { UserProvider };