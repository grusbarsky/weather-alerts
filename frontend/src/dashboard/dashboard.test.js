import React from "react";
import { render } from "@testing-library/react";
import { MemoryRouter } from "react-router";
import Dashboard from "./Dashboard";
import { UserProvider } from "../testUtils";


it("matches snapshot", function () {
  const { asFragment } = render(
      <MemoryRouter>
        <UserProvider>
          <Dashboard />
        </UserProvider>
      </MemoryRouter>,
  );
  expect(asFragment()).toMatchSnapshot();
});
