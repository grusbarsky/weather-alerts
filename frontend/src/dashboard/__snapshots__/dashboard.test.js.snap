// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="container"
  >
    <div
      class="homepage"
    >
      <div
        class="container text-center"
      >
        <div
          class="dashboard container"
        >
          <h1
            id="brand"
          >
            Weather Alerts
          </h1>
          <div
            class="row"
          >
            <div
              class="right col"
            >
              <div
                class="my-locations card shadow my-3"
              >
                <h4
                  class="mt-4"
                >
                  My Locations
                </h4>
                <div
                  class="p-5"
                >
                  <div
                    class="m-5 p-5"
                  >
                    <div
                      class="_loading_overlay_wrapper _loading_overlay_wrapper--active css-ft4slc"
                      data-testid="wrapper"
                    >
                      <div
                        class="_loading_overlay_overlay css-zh5e5g"
                        data-testid="overlay"
                      >
                        <div
                          class="_loading_overlay_content css-42igfv"
                        >
                          <div
                            class="_loading_overlay_spinner css-192sst4"
                          >
                            <svg
                              viewBox="25 25 50 50"
                            >
                              <circle
                                cx="50"
                                cy="50"
                                fill="none"
                                r="20"
                                stroke-miterlimit="10"
                                stroke-width="2"
                              />
                            </svg>
                          </div>
                          Loading...
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
              <div
                class="my-alerts card shadow my-3"
              >
                <h4
                  class="mt-4"
                >
                  My Alerts
                </h4>
                <div
                  class="p-5"
                >
                  <div
                    class="m-5 p-5"
                  >
                    <div
                      class="_loading_overlay_wrapper _loading_overlay_wrapper--active css-ft4slc"
                      data-testid="wrapper"
                    >
                      <div
                        class="_loading_overlay_overlay css-df17o1"
                        data-testid="overlay"
                      >
                        <div
                          class="_loading_overlay_content css-42igfv"
                        >
                          <div
                            class="_loading_overlay_spinner css-192sst4"
                          >
                            <svg
                              viewBox="25 25 50 50"
                            >
                              <circle
                                cx="50"
                                cy="50"
                                fill="none"
                                r="20"
                                stroke-miterlimit="10"
                                stroke-width="2"
                              />
                            </svg>
                          </div>
                          Loading...
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
              <div
                class="my-articles card shadow my-3"
              >
                <h4
                  class="mt-4"
                >
                  My Articles
                </h4>
                <h3
                  class="text-center mt-5"
                >
                  Loading...
                </h3>
              </div>
            </div>
            <div
              class="col"
            >
              <div
                class="articles left card shadow my-3 mh-25"
              >
                <h4
                  class="mt-4"
                >
                  News
                </h4>
                <div
                  class="p-5"
                >
                  <div
                    class="m-5 p-5"
                  >
                    <div
                      class="_loading_overlay_wrapper _loading_overlay_wrapper--active css-ft4slc"
                      data-testid="wrapper"
                    >
                      <div
                        class="_loading_overlay_overlay css-df17o1"
                        data-testid="overlay"
                      >
                        <div
                          class="_loading_overlay_content css-42igfv"
                        >
                          <div
                            class="_loading_overlay_spinner css-192sst4"
                          >
                            <svg
                              viewBox="25 25 50 50"
                            >
                              <circle
                                cx="50"
                                cy="50"
                                fill="none"
                                r="20"
                                stroke-miterlimit="10"
                                stroke-width="2"
                              />
                            </svg>
                          </div>
                          Loading...
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
